### PR TITLE
1041 comprehensive errors

### PR DESCRIPTION
**Pull request by SohailHaqyar**

### CODE REVIEW

File: src/error-details/notification.ts [hunk 1]
```diff
@@ -69,4 +69,46 @@ export const notificationDetailsMap: IDetailsMap = {
...
+  [Details.AccessCodeMissingFromLocals]: {
+    friendlyMessage:
+      'A critical server error has occurred. Please contact support.',
+    technichalReason:
+      'The access code was not found in the response.locals object. Cannot send email!',
+  },
+
+  [Details.FailedToSendPasswordResetEmail]: {
+    friendlyMessage:
+      'A critical server error has occurred. Please contact support.',
```
Comment (zbagley) [inline src/error-details/notification.ts:89]: Unable to deliver the email for a password reset. Try again later or contact support.

File: src/error-details/notification.ts [hunk 2]
```diff
@@ -69,4 +69,46 @@ export const notificationDetailsMap: IDetailsMap = {
...
+    technichalReason:
+      'The email failed to send. This is a server error, and should never happen. Check logs for more information.',
+  },
+
+  [Details.FailedToSendOrganizationAdminInviteEmail]: {
+    friendlyMessage:
+      'A critical server error has occurred. Please contact support.',
```
Comment (zbagley) [inline src/error-details/notification.ts:96]: Organization invite sent and will appear on their dashboard, but an email could not be delivered to the recipient.

File: src/error-details/notification.ts [hunk 3]
```diff
@@ -69,4 +69,46 @@ export const notificationDetailsMap: IDetailsMap = {
...
+    technichalReason:
+      'The email failed to send. This is a server error, and should never happen. Check logs for more information.',
+  },
+
+  [Details.FailedToSendUserRegistrationVerificationEmail]: {
+    friendlyMessage:
+      'A critical server error has occurred. Please contact support.',
```
Comment (zbagley) [inline src/error-details/notification.ts:103]: Unable to deliver the email for registration verification. Try again later or contact support.

Also, we do provide some ability for them to register if this happens without having to wait for their registration to expire + create the account again?

File: src/error-details/notification.ts [hunk 4]
```diff
@@ -69,4 +69,46 @@ export const notificationDetailsMap: IDetailsMap = {
...
+    technichalReason:
+      'The email failed to send. This is a server error, and should almost never happen. Check logs for more information.',
+  },
+
+  [Details.FailedToSendUserRegistrationReverificationEmail]: {
+    friendlyMessage:
+      'A critical server error has occurred. Please contact support.',
```
Comment (zbagley) [inline src/error-details/notification.ts:110]: Unable to deliver the email for registration verification. Try again later or contact support.

**Review by zbagley — CHANGES_REQUESTED** [review]

File: src/error-details/notification.ts (hunk 3 above)
Comment (SohailHaqyar) [reply src/error-details/notification.ts:103]: We do not. If this actually happened right now, the solution would be to mess with the database.

File: src/error-details/notification.ts (hunk 4 above)
Comment (SohailHaqyar) [reply src/error-details/notification.ts]: https://github.com/SalientSys/salient-sst-errors/pull/18/commits/e9da0de5bd05bea12ad2f8ed2073d59813970ddb

File: src/error-details/notification.ts (hunk 3 above)
Comment (SohailHaqyar) [reply src/error-details/notification.ts]: https://github.com/SalientSys/salient-sst-errors/pull/18/commits/e9da0de5bd05bea12ad2f8ed2073d59813970ddb

File: src/error-details/notification.ts (hunk 2 above)
Comment (SohailHaqyar) [reply src/error-details/notification.ts]: https://github.com/SalientSys/salient-sst-errors/pull/18/commits/e9da0de5bd05bea12ad2f8ed2073d59813970ddb

File: src/error-details/notification.ts (hunk 1 above)
Comment (SohailHaqyar) [reply src/error-details/notification.ts]: https://github.com/SalientSys/salient-sst-errors/pull/18/commits/e9da0de5bd05bea12ad2f8ed2073d59813970ddb

**Pull request merged**